src/io/http/src/main/scala/DataFrameClients.scala

```diff
@@ -0,0 +1,150 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import java.util.concurrent.{BlockingQueue, CountDownLatch, LinkedBlockingQueue}
+
+import org.apache.commons.io.IOUtils
+import org.apache.http.client.methods.{CloseableHttpResponse, HttpGet, HttpPost, HttpUriRequest}
+import org.apache.http.entity.StringEntity
+import org.apache.http.impl.client.{CloseableHttpClient, HttpClientBuilder}
+import org.apache.http.message.BufferedHeader
+import org.apache.log4j.{LogManager, Logger}
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.types.{DataType, StringType, StructField, StructType}
+
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+private[ml] trait DataFrameClient extends BaseClient[Row, Row] {
+  def transformSchema(schema: StructType): StructType
+}
+
+private[ml] trait UnaryTransformerClient extends DataFrameClient with ColumnSchema {
+  private var inputIndex: Option[Int] = None
+
+  def setInputIndex(value: Int): Unit = {
+    inputIndex = Some(value)
+  }
+
+  def getInputIndex: Int = inputIndex.get
+
+  private var outputCol: Option[String] = None
+
+  def setOutputCol(value: String): Unit = {
+    outputCol = Some(value)
+  }
+
+  def getOutputCol: String = outputCol.get
+
+  def transformSchema(struct: StructType): StructType = {
+    assert(inputIndex.isDefined)
+    assert(outputCol.isDefined)
+    struct.add(getOutputCol, transformDataType(struct(getInputIndex).dataType))
+  }
+
+}
+
+abstract class UnaryTransformerWrapper[In, Out]
+(val baseClient: BaseClient[In, Out] with ColumnSchema)
+  extends UnaryTransformerClient{
+
+  override def setUrl(value: String): Unit = {
+    baseClient.setUrl(value)
+    super.setUrl(value)
+  }
+
+  override def setPostType(value: String): Unit = {
```

Comment:
rename to `setMethod` or some such thing..

Reply:
Done